integration/ethereum_stellar_test.go

```diff
@@ -471,3 +472,200 @@ func TestStellarToEthereumWithdrawal(t *testing.T) {
+		require.NoError(t, err)
+		err = ethClient.Call(nil, "evm_mine")
+		require.NoError(t, err)
+
+		time.Sleep(time.Second)
+	}
+
+	t.Log("Ethereum time reached withdrawal deadline")
+
+	g := new(errgroup.Group)
+	postData := url.Values{
+		"transaction_hash": {tx.Hash},
+	}
+	txs := make([]string, servers)
+
+	for i := 0; i < servers; i++ {
+		i := i
+		g.Go(func() error {
+			port := 9000 + i
+		loop:
+			for {
+				time.Sleep(time.Second)
+				url := fmt.Sprintf("http://localhost:%d/stellar/refund", port)
+				resp, err := itest.Client().PostForm(url, postData)
+				require.NoError(t, err)
+				switch resp.StatusCode {
+				case http.StatusAccepted:
+					t.Log("Signing request accepted, waiting...")
+					continue loop
+				case http.StatusOK:
+					t.Log("Signing request processed")
+				default:
+					return errors.Errorf("Unknown status code: %s", resp.Status)
+				}
+				body, err := ioutil.ReadAll(resp.Body)
+				require.NoError(t, err)
+				txEnvelope := string(body)
+				txs[i] = txEnvelope
+
+				// Try to submit tx with just one signature
+				_, err = itest.HorizonClient().SubmitTransactionXDR(txEnvelope)
+				require.Error(t, err)
+				return nil
+			}
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Concat signatures
+	gtx, err := txnbuild.TransactionFromXDR(txs[0])
+	require.NoError(t, err)
+	expectedTxHash, err := gtx.HashHex(StandaloneNetworkPassphrase)
+
+	mainTx, ok := gtx.Transaction()
+	require.True(t, ok)
+
+	require.NoError(t, err)
+	require.Equal(t, "3."+strings.Repeat("0", 7), mainTx.Operations()[0].(*txnbuild.Payment).Amount)
+
+	// Add as many sigs as needed and not a single more
+	for i := 1; i < servers/2+1; i++ {
+		gtx, err := txnbuild.TransactionFromXDR(txs[i])
+		require.NoError(t, err)
+		tx, ok := gtx.Transaction()
+		require.True(t, ok)
+
+		// TODO timebound time should be provided in HTTP request and checked by Starbridge
+		txHash, err := gtx.HashHex(StandaloneNetworkPassphrase)
+		require.NoError(t, err)
+		require.Equal(t, expectedTxHash, txHash)
+
+		sig := tx.Signatures()
+
+		mainTx, err = mainTx.AddSignatureDecorated(sig...)
+		require.NoError(t, err)
+	}
+
+	// ...and add client signature (because it's tx source)
+	mainTx, err = mainTx.Sign(StandaloneNetworkPassphrase, itest.clientKey)
+	require.NoError(t, err)
+
+	b64, err := mainTx.Base64()
+	require.NoError(t, err)
+
+	_, err = itest.HorizonClient().SubmitTransaction(mainTx)
+	require.NoErrorf(t, err, "error submitting: %s", b64)
+
+	memoHex := mainTx.ToXDR().Memo().MustHash().HexString()
+	numFound := 0
+	for attempts := 0; attempts < 10; attempts++ {
+		for i := 0; i < servers; i++ {
+			found, err := stores[i].HistoryStellarTransactionExists(context.Background(), memoHex)
+			require.NoError(t, err)
+			if found {
+				numFound++
+			}
+		}
+		if numFound == servers {
+			break
+		} else {
+			numFound = 0
+			time.Sleep(time.Second)
+		}
+	}
+	require.Equal(t, servers, numFound)
```

Comment:
can you try to repeat the http request for the refund endpoint and assert that the request fails with a refund already executed error?